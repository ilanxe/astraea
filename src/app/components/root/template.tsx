import * as React from 'react'
import {PAGE_STYLE} from './styles'

/**
 * Render root component representing the entire application
 * @return Root component
 */
export default function Root() {
  return <div style={PAGE_STYLE}>Hello, World</div>
}
