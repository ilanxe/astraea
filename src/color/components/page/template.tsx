import * as React from 'react'
import styled from 'styled-components'

// Background component
interface BackgroundProps {
  color: string
  className?: string
  children?: React.ReactNode
}
let Background = ({children, className}: BackgroundProps) => (
  <div className={className}>{children}</div>
)
const StyledBackground = styled(Background)`
  transition: background 0.8s ease-out;
  display: flex;
  background-color: ${({color}) => color};
`

// Text component
const Text = styled.span`
  margin-left: 4px;
  margin-right: 4px;
`

// Gradient overlay component
const Gradient = styled.div`
  flex: 1;
  display: flex;
  justify-content: center;
  align-items: center;
  background: linear-gradient(rgba(255, 255, 255, 0.4), transparent);
`

/** Button component */
export const Button = styled.button`
  border-radius: 50%;
  background: radial-gradient(white, gainsboro);
  border-color: gainsboro;
  outline: none;
`

/** Properties for color page component */
export interface Props {
  /** CSS color to show */
  readonly color: string

  /** Additional class names to include */
  readonly className?: string
}

/** Actions for color page component */
export interface Actions {
  /** Handler when user wants to go to next color */
  onNextColor(): void

  /** Handler when user wants to go to previous color */
  onPreviousColor(): void
}

/**
 * Render color page component showing current color and buttons to change
 * @param props Component properties
 * @return Color page component
 */
export default function ColorPage({
  className,
  color,
  onNextColor,
  onPreviousColor,
}: Props & Actions) {
  return (
    <StyledBackground color={color} className={className}>
      <Gradient>
        <Button onClick={onPreviousColor}>&lt;</Button>
        <Text>Hello, World</Text>
        <Button onClick={onNextColor}>&gt;</Button>
      </Gradient>
    </StyledBackground>
  )
}
