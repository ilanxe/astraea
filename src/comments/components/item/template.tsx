import * as React from 'react'
import {DataProps} from 'react-apollo'
import * as Wrapper from '.'

/** Post item properties */
export type Props = DataProps<Wrapper.Data>

/**
 * Render comment item detail
 * @return Comment item component
 */
export function CommentsItem({data: {loading, comment}}: Props) {
  if(loading) {
    return <>Loading</>
  }
  if(comment === undefined) {
    return <>Error</>
  }
  const comments = comment.comments.map(c => (
    <div key={c.id}><Wrapper.CommentsItem {...c} /></div>
  ))
  return (
    <>
      <div>{comment.author.name}</div>
      <div>{comment.text}</div>
      {comments}
    </>
  )
}
