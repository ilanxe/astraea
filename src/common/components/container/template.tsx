import * as React from 'react'
import {Provider} from 'react-redux'
import {Store} from 'redux'
const {AppContainer} = require('react-hot-loader')

/** Container component properties */
interface Props<P> {
  /** Redux store instance */
  readonly store: Store<any> // tslint:disable-line:no-any

  /** Component to render */
  readonly component: React.ComponentClass<P> | (() => React.ReactElement<P>)
}

/**
 * Render container component adding possible dev tools and redux store
 * @param props Component properties
 * @return Container component
 */
export default function Container<P>({store, component: Component}: Props<P>) {
  return (
    <AppContainer>
      <Provider store={store}>
        <Component />
      </Provider>
    </AppContainer>
  )
}
