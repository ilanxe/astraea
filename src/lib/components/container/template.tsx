import * as React from 'react'
import {Provider} from 'react-redux'
import {Store} from 'redux'
const {AppContainer} = require('react-hot-loader')

/** Container component properties */
interface Props {
  /** Redux store instance */
  store: Store<any> // tslint:disable-line:no-any

  /** Component to render */
  component: any // tslint:disable-line:no-any
}

/**
 * Render container component adding possible dev tools and redux store
 * @param props Component properties
 * @return Container component
 */
export default function Container({store, component: Component}: Props) {
  return (
    <AppContainer>
      <Provider store={store}>
        <Component />
      </Provider>
    </AppContainer>
  )
}
