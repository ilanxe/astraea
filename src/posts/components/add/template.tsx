import * as React from 'react'
import {Field, InjectedFormProps} from 'redux-form'
import {Post} from '../../graphql'

/** Form attributes */
export type Form = Partial<Pick<Post, 'description' | 'title'>>

/**
 * Render add post form component
 * @return Add post component
 */
export function PostsAdd({
  handleSubmit,
  pristine,
  reset,
  submitting,
  valid,
}: InjectedFormProps<Form>) {
  const disableClear = pristine || submitting
  const disableSubmit = submitting || !valid
  return (
    <form onSubmit={handleSubmit}>
      <div>
        <label>
          Title:
          <Field name="title" component="input" type="text" />
        </label>
      </div>
      <div>
        <label>
          Description:
          <Field name="description" component="input" type="text" />
        </label>
      </div>
      <div>
        <button type="button" disabled={disableClear} onClick={reset}>
          Clear
        </button>
        <button type="submit" disabled={disableSubmit}>Submit</button>
      </div>
    </form>
  )
}
