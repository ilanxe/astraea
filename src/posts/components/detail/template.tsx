import * as React from 'react'
import {Post} from '../../graphql'

export const NO_POST: Post = {
  id: 0,
  title: '',
  description: '',
  author: undefined!,
  comments: [],
}

/** Interfaces for PostsList component properties */
export namespace PostsDetail {
  /** Component data */
  export interface Data {
    post: Post
  }
  /** Component properties */
  export type Props = Data
}

/** Render list of posts */
export function PostsDetail({post}: PostsDetail.Props) {
  return <div>{post.title}</div>
}
