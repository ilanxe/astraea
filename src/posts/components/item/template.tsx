import * as React from 'react'
import {DataProps} from 'react-apollo'
import {Data} from '.'

/** Post item properties */
export type Props = DataProps<Data>

/**
 * Render post item detail
 * @return Post item component
 */
export function PostsItem({data: {loading, post}}: Props) {
  if(loading) {
    return <>Loading</>
  }
  if(post === undefined) {
    return <>Error</>
  }
  return (
    <>
      <div>{post.title}</div>
      <div>{post.description}</div>
    </>
  )
}
