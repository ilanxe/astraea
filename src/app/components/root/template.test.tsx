import {shallow} from 'enzyme'
import * as React from 'react'
import {AppRoot} from './template'

describe('<AppRoot> Template', () => {
  it('should render as expected', () => {
    expect(shallow(<AppRoot component={FauxComponent} />)).toMatchSnapshot()
  })
})

function FauxComponent() {
  return <div>Hello, world</div>
}
