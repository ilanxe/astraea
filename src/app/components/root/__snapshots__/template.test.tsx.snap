// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppRoot> Template should render as expected 1`] = `
<div>
  <div>
    <Connect(Link)
      to={
        Object {
          "type": "COMMON_NAVIGATE_HOME",
        }
      }
    >
      Home
    </Connect(Link)>
    |
    <Connect(Link)
      to={
        Object {
          "type": "POSTS_NAVIGATE_ADD",
        }
      }
    >
      Add Post
    </Connect(Link)>
  </div>
  <FauxComponent />
</div>
`;
